compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -255,7 +257,8 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
                   else tp.baseTypeWithArgs(cls)
                 base.mapReduceOr(identity)(mergeRefined)
               }
-              doms.map(baseTp).reduceLeft(AndType.apply)
+              if (doms.isEmpty) new ErrorType("no parents in common") // This can happen in the union of Any with PhantomAny
```

Comment:
This currently happens after while typing `if` where one branch is phantom and the other is not. In these cases the error has already been reported but typing continues. I should be able to avoid this while typing the `if`.